src/tool/GenerateAction.cpp

```diff
@@ -74,12 +120,21 @@ DoGenerateAction()
     // Load the compilation database file
     //
     // --------------------------------------------------------------
+    
     MRDOCS_CHECK(toolArgs.inputPaths, "The compilation database path argument is missing");
     MRDOCS_CHECK(toolArgs.inputPaths.size() == 1,
         formatError(
             "got {} input paths where 1 was expected",
             toolArgs.inputPaths.size()));
-    auto compilationsPath = files::normalizePath(toolArgs.inputPaths.front());
+
+    auto const inputPath = generateCompilationDatabase(toolArgs.inputPaths.front(), config->settings().cmake);
```

Comment:
This function name is ambiguous. It could be just generateCompileCommandsFile or something like that. CompilationDatabase here usually refers to an LLVM object type that can be generated in many ways (including reading a compile_commands.json file).

Reply:
Instead of `config->settings().cmake`, you can just read that property from the object if it exists or provide an empty string otherwise.